### PR TITLE
Docs on URL components.

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java [hunk 1]
```diff
@@ -29,13 +29,108 @@
...
+ *   }
+ * }</pre>
+ *
+ * which prints: <pre>   {@code
+ *
+ *   q: cute #puppies
+ *   f: images
+ * }</pre>
+ *
+ * In addition to composing URLs from their component parts, and decomposing URLs into their
```
Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: I think the comma isn't needed

File: okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java [hunk 2]
```diff
@@ -29,13 +29,108 @@
...
+ *   https://www.youtube.com/watch?v=cbP2N1BQdYc
+ * }</pre>
+ *
+ * <h3>What's in a URL?</h3>
+ *
+ * A URL has several components.
+ *
+ * <h4>Scheme</h4>
+ * Sometimes referred to as <i>protocol</i>, A URL's scheme describes what mechanism should be used
+ * to retrieve the resource. Although URLs have many schemes (mailto, file, ftp), this class only
```
Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: These schemes are not wrapped on code while http and https are. Normalize?

**Comment by JakeWharton** [comment]
Nice oxford comma usage!

**Comment by swankjesse** [comment]
https://github.com/square/okhttp/issues/1486

**Pull request merged**